src/meta/src/controller/fragment.rs

```diff
@@ -700,19 +713,32 @@ impl CatalogController {
         )
     }
 
-    pub async fn list_streaming_job_states(
-        &self,
-    ) -> MetaResult<Vec<(ObjectId, JobStatus, StreamingParallelism, i32)>> {
+    pub async fn list_streaming_job_infos(&self) -> MetaResult<Vec<StreamingJobInfo>> {
         let inner = self.inner.read().await;
         let job_states = StreamingJob::find()
             .select_only()
+            .column(streaming_job::Column::JobId)
+            .join(JoinType::InnerJoin, streaming_job::Relation::Object.def())
+            .column(object::Column::ObjType)
+            .join(JoinType::LeftJoin, table::Relation::Object1.def().rev())
+            .join(JoinType::LeftJoin, source::Relation::Object.def().rev())
+            .join(JoinType::LeftJoin, sink::Relation::Object.def().rev())
+            .column_as(
+                Expr::if_null(
+                    Expr::col((table::Entity, table::Column::Name)),
+                    Expr::if_null(
+                        Expr::col((source::Entity, source::Column::Name)),
+                        Expr::col((sink::Entity, sink::Column::Name)),
+                    ),
+                ),
+                "name",
```

Comment:
Seems no better way to obtain the job name. 🤡

Reply:
It's a left join, so I guess all sources queried here are shared sources?

Reply:
🥵 All streaming jobs will be listed, including those being `Creating`. Is this by design? Add FYI for replace table we will have a dummy record in `streaming_job` table that doesn't have any associated catalogs in other tables.

Reply:
> including those being `Creating`

I think this can be useful for debugging and is frequently asked: https://github.com/risingwavelabs/risingwave/issues/19044

Reply:
> for replace table we will have a dummy record

This is just a temporary state I suppose, so perhaps it won't be too much problem. 🥺

Reply:
I guess it will cause panic because the `name` will be `NULL` and can't remapped to `String` in partial model `StreamingJobInfo`.

Reply:
Not sure if the following filter works or not. You can give it a try.
```rust
.filter(Expr::col(Alias::new("name")).is_not_null())
```

Reply:
Added a fallback of `<unknown>`